sdk/cosmos/azure-cosmos/test/test_crud.py

```diff
@@ -567,8 +567,8 @@ def test_partitioned_collection_permissions(self):
 
         resource_tokens = {}
         # storing the resource tokens based on Resource IDs
-        resource_tokens[urllib.quote(all_collection.id)] = (all_permission.properties['_token'])
-        resource_tokens[urllib.quote(read_collection.id)] = (read_permission.properties['_token'])
+        resource_tokens["dbs/" + created_db.id + "/colls/" + all_collection.id] = (all_permission.properties['_token'])
+        resource_tokens["dbs/" + created_db.id + "/colls/" + read_collection.id] = (read_permission.properties['_token'])
```

Comment:
Was the existing test invalid in some way?
Rather than alter existing tests, we should simply add new ones - we don't want to inadvertently break existing customers. All the existing tests should pass without modification (unless the tests were never correct in the first place).

Reply:
So the way in which resource tokens were implemented were faulty. resource tokens are supposed to be a map of resource link to token . i.e {"dbs/mydb/coll/mycoll1": "token1", "dbs/mydb/colls/mycoll2": "token2"} (this is the implementation in the .Net SDK and is the spec as well). But in the python sdk currently, we expect to get it to be a map of resource id to token. i.e {"mycoll1": "token1", "mycoll2": "token2"}. This is faulty since the same collection name can exist in 2 different databases.
These tests also follow the old pattern. They will break with the new changes, so I changed the tests as well.

Reply:
Thanks, but is it possible any customers were using the old faulty pattern? For example by manually building this dictionary? Or if they had done that would it not have worked?

Reply:
@annatisch It is possible that some cases would have worked but that is not intentional. We consider it a bug and fixed the same issue in JS SDK recently. 

Previous versions of the Cosmos SDK were all based on `rid` which was a unique identifier resource. The transition to user supplied ids a couple years ago changed that, but we didn't fix resource tokens to account for it. Given all this, and the fact that resource tokens do not see a lot of usage, I am fine with merging this as a bugfix.

Reply:
I would still prefer this to be backwards compatible - but if you've rolled out the same change in other languages I guess that's okay.